### PR TITLE
Vapor 4

### PR DESCRIPTION
**Pull request by mcdappdev**

### CODE REVIEW

**Comment by mcdappdev** [comment]
Oops, Xcode was being silly and didn't tell me about a bunch of errors because of a bad import statement. Will fix and repush

File: Sources/Jobs/JobsCommand.swift [hunk 1]
```diff
@@ -31,18 +41,26 @@ public class JobsCommand: Command {
...
+        self.jobContext = jobContext
+        self.config = config
     }
     
     /// See `Command`.`run(using:)`
-    public func run(using context: CommandContext) throws -> EventLoopFuture<Void> {
+    public func run(using context: CommandContext<JobsCommand>) throws {
         context.console.info("Starting Jobs worker")
-
+        let queueName = try context.option(\Signature.queue) ?? QueueName.default.name
```
Comment (tanner0101) [inline Sources/Jobs/JobsCommand.swift]: You can do just `\.queue` here

File: Sources/Jobs/JobsCommand.swift [hunk 2]
```diff
@@ -66,68 +84,61 @@ public class JobsCommand: Command {
...
             }
         }
         
-        return .andAll(shutdownPromises.map { $0.futureResult }, eventLoop: elg.next())
+        try EventLoopFuture.andAllComplete(shutdownPromises.map { $0.futureResult }, on: elg.next()).wait()
     }
     
     private func setupTask(eventLoop: EventLoop,
-                           container: SubContainer,
                            queueName: String,
```
Comment (tanner0101) [inline Sources/Jobs/JobsCommand.swift]: Xcode formatting

File: Sources/Jobs/JobsCommand.swift [hunk 3]
```diff
@@ -66,68 +84,61 @@ public class JobsCommand: Command {
...
                            promise: EventLoopPromise<Void>) throws
     {
         let queue = QueueName(name: queueName)
-        let queueService = try container.make(QueueService.self)
-        let jobContext = try container.make(JobContext.self)
         let key = queue.makeKey(with: queueService.persistenceKey)
-        let config = try container.make(JobsConfig.self)
-        
-        _ = eventLoop.scheduleRepeatedTask(initialDelay: .seconds(0), delay: queueService.refreshInterval) { task -> EventLoopFuture<Void> in
+        _ = eventLoop.scheduleRepeatedAsyncTask(initialDelay: .seconds(0), delay: queueService.refreshInterval) { task -> EventLoopFuture<Void> in
```
Comment (tanner0101) [inline Sources/Jobs/JobsCommand.swift]: while we're here, should we try to handle this future better? the least we could do is print something if it fails

File: Sources/Jobs/JobsProvider.swift [hunk 4]
```diff
@@ -26,18 +27,12 @@ public struct JobsProvider: Provider {
     
     /// See `Provider`.`register(_ services:)`
     public func register(_ services: inout Services) throws {
-        services.register { container -> QueueService in
+        services.register(QueueService.self) { container -> QueueService in
             let persistenceLayer = try container.make(JobsPersistenceLayer.self)
             
             return QueueService(refreshInterval: self.refreshInterval,
```
Comment (tanner0101) [inline Sources/Jobs/JobsProvider.swift]: Xcode formatting

**Comment by penny-coin** [comment]
Hey @mcdappdev, you just merged a pull request, have a coin! 

You now have 303 coins.

File: Sources/Jobs/JobsConfig.swift [hunk 5]
```diff
@@ -2,7 +2,7 @@ import Foundation
 import Vapor
 
 /// A `Service` to configure `Job`s
-public struct JobsConfig: Service {
+public struct JobsConfig {
```
Comment (tanner0101) [inline Sources/Jobs/JobsConfig.swift]: We should rename this to `JobsConfiguration`

**Pull request merged**